### PR TITLE
Make many more variables 'const'.

### PR DESCRIPTION
**Pull request by bangerth**
Following up on #16942 and #16945, I thought I'd look around and see via script whether there are other opportunities to make variables `const`. This primarily addresses quadrature objects, though if I had to edit something by hand, I also made surrounding variable const where possible. It's clear that there are many many more opportunities to do this, but that may be for another day.

### CODE REVIEW

File: examples/step-35/step-35.cc [hunk 1]
```diff
@@ -451,8 +451,8 @@ namespace Step35
     DoFHandler<dim> dof_handler_velocity;
     DoFHandler<dim> dof_handler_pressure;
 
-    QGauss<dim> quadrature_pressure;
-    QGauss<dim> quadrature_velocity;
+    const QGauss<dim> quadrature_pressure;
+    const QGauss<dim> quadrature_velocity;
```
Comment (drwells) [inline examples/step-35/step-35.cc:455]: This doesn't look right since this is a declaration.

File: examples/step-72/step-72.cc [hunk 2]
```diff
@@ -159,9 +159,9 @@ namespace Step72
 
     Triangulation<dim> triangulation;
 
-    DoFHandler<dim> dof_handler;
-    FE_Q<dim>       fe;
-    QGauss<dim>     quadrature_formula;
+    DoFHandler<dim>   dof_handler;
+    const FE_Q<dim>   fe;
+    const QGauss<dim> quadrature_formula;
```
Comment (drwells) [inline examples/step-72/step-72.cc:164]: Same

**Review by drwells — CHANGES_REQUESTED** [review]

File: examples/step-35/step-35.cc (hunk 1 above)
Comment (bangerth) [reply examples/step-35/step-35.cc:455]: It's a member variable. It's going to be initialized in the constructor, but it can be `const`, no?

Comment (drwells) [reply examples/step-35/step-35.cc:455]: you're right!

**Review by drwells — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Pull request merged**